client/RequestPacker.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <iostream>
+#include <vector>
+#include <stdexcept>
+#include <string>
+#include <cstring> // for memcpy
+
+// Base class for packing request packets
+class RequestPacketPacker {
+public:
+    // Constructor to initialize RequestPacketPacker fields
+    RequestPacketPacker(const std::vector<uint8_t>& clientId, uint8_t version, uint16_t code, const std::vector<uint8_t>& payload);
+
+    // Virtual function to pack the packet
+    virtual std::vector<uint8_t> pack() const;
+
+protected:
+    uint8_t client_id[16];
+    uint8_t version;
+    uint16_t code;
+    std::vector<uint8_t> payload;

```

Comment:
wy are those protected? and why is there a payload field?
the whole point of this inheritence is that every layer contains its own data and itll be parse during object construction. You're close but youre missing the main point of this design pattern